cabal-install/main/Main.hs

```diff
@@ -445,40 +451,80 @@ reconfigureAction flags@(configFlags, _) _ globalFlags = do
-                    (buildOnly buildExFlags)
-  (useSandbox, config) <- loadConfigOrSandboxConfig verbosity globalFlags
-  distPref <- findSavedDistPref config (buildDistPref buildFlags)
-  -- Calls 'configureAction' to do the real work, so nothing special has to be
-  -- done to support sandboxes.
-  config' <-
-    reconfigure configureAction
-    verbosity distPref useSandbox noAddSource (buildNumJobs buildFlags)
-    mempty [] globalFlags config
-  nixShell verbosity distPref globalFlags config $ do
-    maybeWithSandboxDirOnSearchPath useSandbox $
-      build verbosity config' distPref buildFlags extraArgs
 
+buildAction :: (BuildFlags, BuildExFlags) -> [String] -> Action
+buildAction flags@(buildFlags, _) = buildActionForCommand
+  (Cabal.buildCommand defaultProgramDb)
+  verbosity
+  flags
+  where verbosity = fromFlagOrDefault normal (buildVerbosity buildFlags)
+
+showBuildInfoAction :: (BuildFlags, BuildExFlags) -> [String] -> Action
+showBuildInfoAction flags@(buildFlags, _) = buildActionForCommand
+  (Cabal.showBuildInfoCommand defaultProgramDb)
+  verbosity
+  flags
+    -- Default silent verbosity so as not to pollute json output
+  where verbosity = fromFlagOrDefault silent (buildVerbosity buildFlags)
+
+buildActionForCommand :: CommandUI BuildFlags
+                      -> Verbosity
+                      -> (BuildFlags, BuildExFlags)
+                      -> [String]
+                      -> Action
+buildActionForCommand commandUI verbosity (buildFlags, buildExFlags) extraArgs globalFlags
+  = do
+    let noAddSource =
+          fromFlagOrDefault DontSkipAddSourceDepsCheck (buildOnly buildExFlags)
+    (useSandbox, config) <- loadConfigOrSandboxConfig verbosity globalFlags
+    distPref             <- findSavedDistPref config (buildDistPref buildFlags)
+    -- Calls 'configureAction' to do the real work, so nothing special has to be
+    -- done to support sandboxes.
+    config'              <- reconfigure configureAction
+                                        verbosity
+                                        distPref
+                                        useSandbox
+                                        noAddSource
+                                        (buildNumJobs buildFlags)
+                                        mempty
+                                        []
+                                        globalFlags
+                                        config
+    nixShell verbosity distPref globalFlags config $ do
+      maybeWithSandboxDirOnSearchPath useSandbox $ buildForCommand commandUI
+                                                                   verbosity
+                                                                   config'
+                                                                   distPref
+                                                                   buildFlags
+                                                                   extraArgs
```

Comment:
Please keep the code style intact, this just looks horrid IMO.

Something like:

```
    nixShell verbosity distPref globalFlags config $ do
      maybeWithSandboxDirOnSearchPath useSandbox $
        buildForCommand commandUI verbosity config' distPref buildFlags extraArgs
```

or if that is >80chars:

```
    nixShell verbosity distPref globalFlags config $ do
      maybeWithSandboxDirOnSearchPath useSandbox $ buildForCommand
       commandUI verbosity config' distPref buildFlags extraArgs
```

Reply:
Is there a reference to your coding style? Also maybe a configured formatter?

Reply:
I don't think Cabal has a canonical coding style document, though I have never really looked. In general it's a good idea to apply the surrounding coding style so as to not make the styling inconsistent. Personally I just do formatting by hand so I can't really help you with a config.

Reply:
See https://github.com/haskell/cabal#conventions. We're loosely following Johan Tibell's [style guide](https://github.com/tibbe/haskell-style-guide/blob/master/haskell-style.md).